cpp/src/utilities/cuda_utils.hpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 BlazingDB, Inc.
+ *     Copyright 2019 Eyal Rozenberg <eyalroz@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CUDF_UTILITIES_CUDA_UTIL_CUH_
+#define CUDF_UTILITIES_CUDA_UTIL_CUH_
+
+/**
+ * @file Utility code for host-side interaction with CUDA by cuDF API
+ * function implementations
+ */
+
+#include <utilities/integer_utils.hpp>
+
+namespace cudf {
+
+namespace util {
+
+namespace cuda {
+
+inline constexpr auto form_naive_1d_grid(
```

Comment:
The naming leaves something to be desired here. Also, Doxygen with a usage example would be nice.

Reply:
Do you have a specific idea regarding naming? How about "simple 1d grid"?

Reply:
I've redone this as a class instead of a function returning an anonymous struct. Also, renamed it. Tell me what you think.